src/aiperf/common/config/input_config.py

```diff
@@ -98,6 +98,39 @@ def validate_custom_dataset_file(self) -> Self:
             raise ValueError("Custom dataset type requires --input-file to be provided")
         return self
 
+    @model_validator(mode="after")
+    def validate_rankings_token_options(self) -> Self:
+        """Validate that prompt input tokens and rankings-specific token options are not both set."""
+        from aiperf.common.config.config_defaults import InputTokensDefaults
+
+        # Check if prompt input tokens have been changed from defaults
+        prompt_tokens_modified = (
+            self.prompt.input_tokens.mean != InputTokensDefaults.MEAN
+            or self.prompt.input_tokens.stddev != InputTokensDefaults.STDDEV
+        )
+
+        # Check if any rankings-specific token options have been changed from defaults
+        rankings_tokens_modified = (
+            self.rankings_passages_prompt_token_mean
+            != InputDefaults.RANKINGS_PASSAGES_PROMPT_TOKEN_MEAN
+            or self.rankings_passages_prompt_token_stddev
+            != InputDefaults.RANKINGS_PASSAGES_PROMPT_TOKEN_STDDEV
+            or self.rankings_query_prompt_token_mean
+            != InputDefaults.RANKINGS_QUERY_PROMPT_TOKEN_MEAN
+            or self.rankings_query_prompt_token_stddev
+            != InputDefaults.RANKINGS_QUERY_PROMPT_TOKEN_STDDEV
+        )
+
+        if prompt_tokens_modified and rankings_tokens_modified:
```

Comment:
Also you will want to decide whether you want the rankings endpoints to fallback to using the prompt size if not set. 


If yes, you want to fallback when not set, then apply here (or in the rankings logic itself)

If no, then you should fail if the user added prompt token stuff at all and is using rankings:

(note i cant remember the exact variable names)

```python
if "rankings" in endpoint.type and prompt_tokens_modified:
   raise("Whatchua talking about willis?")
```
or similar

<img width="640" height="640" alt="Image" src="https://github.com/user-attachments/assets/505c003e-23e2-48ff-bf3f-f4886d69e9b9" />

Reply:
I went for the more aggressive approach.
If the endpoint type is rankings: cohere, hf tei, or nim, then prompt token cannot be set.
This validation is in the user_config.